core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -108,6 +115,32 @@ class DefaultAutoTopicCreationManager(
     uncreatableTopicResponses ++ creatableTopicResponses
   }
 
+  override def createStreamsInternalTopics(
+    topics: Map[String, CreatableTopic],
+    requestContext: RequestContext
+  ): Unit = {
+
+    // Set default values for numPartitions and replicationFactor if they are not provided
```

Comment:
Really needed?

Reply:
Assuming you mean the inline comment. Removed it.